### PR TITLE
Add routing info endpoint

### PR DESCRIPTION
**Pull request by klapkov**
This PR is related to this issue: https://github.com/cloudfoundry/diego-release/issues/678. As discussed, here we introduce a new endpoint in the bbs - routing_info - which the route emitter can use when syncing to retrieve a desiredLRP that includes only the information the route emitter needs. 

In cases where there are a huge amount of security groups in a single space and multiple apps deployed in that space we notice that the network traffic consistently increases as more apps are added to a space. After investigation we found that it was due to the large amount of security groups being sent to the route emitters. 

With this approach, we see a great drop in network load when we do have such large spaces and a lot of security groups. We do not introduce a new structure of the desiredLRP as agreed previously, but just return a desiredLRP with only these fields set:

- process_guid
- domain
- log_guid
- instances
- routes
- modification_tag_epoch
- modification_tag_index
- metric_tags 

The run_info is still retrieved and decrypted since we need the metric_tags that are in it, but only the tags are returned. 

One thing to mention is that this route will be used by the route emitter only when we sync - every 60s, which still brings great improvements, but in all other cases, we still receive the desiredLRP trough the event emitted by BBS.

### CODE REVIEW

**Comment by klapkov** [comment]
Related PR in the route emitter: https://github.com/cloudfoundry/route-emitter/pull/23

File: client.go [hunk 1]
```diff
@@ -599,6 +602,20 @@ func (c *client) DesiredLRPSchedulingInfos(logger lager.Logger, filter models.De
 	return response.DesiredLrpSchedulingInfos, response.Error.ToError()
 }
 
+func (c *client) DesiredLRPRoutingInfos(logger lager.Logger, filter models.DesiredLRPFilter) ([]*models.DesiredLRP, error) {
+	request := models.DesiredLRPsRequest{
+		Domain:       filter.Domain,
```
Comment (PlamenDoychev) [inline client.go:607]: Perhaps 'Domain' is not required here as we request filtering based on guid only in route-emitter

File: db/sqldb/desired_lrp_db.go [hunk 2]
```diff
@@ -205,6 +205,60 @@ func (db *SQLDB) DesiredLRPSchedulingInfos(ctx context.Context, logger lager.Log
...
 
+func (db *SQLDB) DesiredLRPRoutingInfos(ctx context.Context, logger lager.Logger, filter models.DesiredLRPFilter) ([]*models.DesiredLRP, error) {
+	logger = logger.Session("db-desired-lrps-routing-infos", lager.Data{"filter": filter})
+	logger.Debug("starting")
+	defer logger.Debug("complete")
+
+	var wheres []string
+	var values []interface{}
+
+	if filter.Domain != "" {
```
Comment (PlamenDoychev) [inline db/sqldb/desired_lrp_db.go:216]: Perhaps this 'if' statement is not needed as we do not pass 'Domain' as filtering element in route-emitter

File: db/sqldb/desired_lrp_db_test.go [hunk 3]
```diff
@@ -329,6 +329,77 @@ var _ = Describe("DesiredLRPDB", func() {
 		})
 	})
 
+	FDescribe("DesiredLRPRoutingInfos", func() {
```
Comment (PlamenDoychev) [inline db/sqldb/desired_lrp_db_test.go:332]: Probably forgotten fix describe 'FDescribe'

File: db/sqldb/desired_lrp_db_test.go [hunk 4]
```diff
@@ -329,6 +329,77 @@ var _ = Describe("DesiredLRPDB", func() {
...
+		})
+
+		It("returns all desired lrps routing infos", func() {
+			desiredLRPRoutingInfos, err := sqlDB.DesiredLRPRoutingInfos(ctx, logger, models.DesiredLRPFilter{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(desiredLRPRoutingInfos).To(HaveLen(3))
+			Expect(desiredLRPRoutingInfos).To(ConsistOf(expectedDesiredLRPRoutingInfos))
+		})
+
+		Context("when filtering by domain", func() {
```
Comment (PlamenDoychev) [inline db/sqldb/desired_lrp_db_test.go:362]: Do we need this context at all as we do not request filtering by 'Domain'?

**Review by PlamenDoychev — CHANGES_REQUESTED** [review]

**Comment by klapkov** [comment]
@PlamenDoychev Removed the domain from the desiredLRP filter. You are right, we only filter by process_guid. Thanks

**Comment by geofffranks** [comment]
@klapkov @PlamenDoychev is there any more work to be done on this PR?

**Comment by klapkov** [comment]
@geofffranks Nope, this is all for this PR.

**Comment by geofffranks** [comment]
Cool! If you're ready to switch your review status to approved @PlamenDoychev I'll merge this in tomorrow.

**Comment by mariash** [comment]
Given that all the questions were resolved we merged this with @reneighbor because route-emitter needed this and it was blocking our CI.

**Comment by PlamenDoychev** [comment]
Thanks @mariash @geofffranks !

Sorry we had bank holiday yesterday and did not manage to react.

**Pull request merged**